File: myExpenses/src/main/java/org/totschnig/myexpenses/preference/PreferenceUtils.java
package org.totschnig.myexpenses.preference;

public class PreferenceUtils {
  private PreferenceUtils() {
  }

  public static boolean shouldStartAutoFill() {
    return PrefKey.AUTO_FILL_AMOUNT.getBoolean(false) ||
        PrefKey.AUTO_FILL_CATEGORY.getBoolean(false) ||
        PrefKey.AUTO_FILL_COMMENT.getBoolean(false) ||
        PrefKey.AUTO_FILL_METHOD.getBoolean(false) ||
        !PrefKey.AUTO_FILL_ACCOUNT.getString("never").equals("never");
  }

  public static void enableAutoFill() {
    PrefKey.AUTO_FILL_AMOUNT.putBoolean(true);
    PrefKey.AUTO_FILL_CATEGORY.putBoolean(true);
    PrefKey.AUTO_FILL_COMMENT.putBoolean(true);
    PrefKey.AUTO_FILL_METHOD.putBoolean(true);
    PrefKey.AUTO_FILL_ACCOUNT.putString("aggregate");
  }

  public static void disableAutoFill() {
    PrefKey.AUTO_FILL_AMOUNT.putBoolean(false);
    PrefKey.AUTO_FILL_CATEGORY.putBoolean(false);
    PrefKey.AUTO_FILL_COMMENT.putBoolean(false);
    PrefKey.AUTO_FILL_METHOD.putBoolean(false);
    PrefKey.AUTO_FILL_ACCOUNT.putString("never");
  }
}

File: myExpenses/src/main/java/org/totschnig/myexpenses/preference/PreferenceUtils.kt
package org.totschnig.myexpenses.preference

fun shouldStartAutoFill(prefHandler: PrefHandler) = with(prefHandler) {
    getBoolean(PrefKey.AUTO_FILL_SWITCH, false) &&
            (getBoolean(PrefKey.AUTO_FILL_AMOUNT, true)
                    || getBoolean(PrefKey.AUTO_FILL_CATEGORY, true)
                    || getBoolean(PrefKey.AUTO_FILL_COMMENT, true)
                    || getBoolean(PrefKey.AUTO_FILL_METHOD, true)
                    || getString(PrefKey.AUTO_FILL_ACCOUNT, "aggregate") != "never")
}

fun shouldStartAutoFillWithFocus(prefHandler: PrefHandler) = with(prefHandler) {
    shouldStartAutoFill(prefHandler) && getBoolean(PrefKey.AUTO_FILL_FOCUS, true)
}

fun enableAutoFill(prefHandler: PrefHandler) {
    prefHandler.putBoolean(PrefKey.AUTO_FILL_SWITCH, true)
    prefHandler.putBoolean(PrefKey.AUTO_FILL_HINT_SHOWN, true)
}

fun disableAutoFill(prefHandler: PrefHandler) =
        prefHandler.putBoolean(PrefKey.AUTO_FILL_SWITCH, false)
